heidrun/digcomm_mods.rb

```diff
@@ -0,0 +1,289 @@
+    'matrix-number', 'music-plate', 'music-publisher',
+    'sici', 'videorecording-identifier'
+  ]
+
+  types.map do |identifier_type|
+    r['mods:identifier']
+      .match_attribute(:type, identifier_type)
+      .map do |i|
+        format('%s: %s',
+               identifier_type.tr('-', ' ').capitalize,
+               i.value)
+      end
+  end
+end
+
+# <mods:location><mods:physicalLocation>
+# CONCATENATED with ". "
+# <mods:location><mods:holdingSimple><mods:copyInformation><mods:subLocation>
+# CONCATENATED with ". "
+# <mods:relatedItem type="host"><mods:titleInfo><mods:title>
+# CONCATENATED with ". "
+# <mods:relatedItem type="series"><mods:titleInfo><mods:title>
+#
+# (e.g., Boston Public Library. Leslie Jones photograph collection)
+build_relation = lambda do |r|
+  field_values = []
+
+  field_values << r['mods:location']
+                 .fields(['mods:physicalLocation'],
+                         ['mods:holdingSimple',
+                          'mods:copyInformation',
+                          'mods:subLocation'])
+
+  field_values << r['mods:relatedItem'].match_attribute(:type, 'host')
+                 .field('mods:titleInfo', 'mods:title')
+
+  field_values << r['mods:relatedItem'].match_attribute(:type, 'series')
+                 .field('mods:titleInfo', 'mods:title')
+
+  field_values.map(&:values).flatten.join('. ')
+end
+
+Krikri::Mapper.define(:digcomm_mods, :parser => Krikri::ModsParser) do
+  provider :class => DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/digital-commonwealth'
+    label 'Digital Commonwealth'
+  end
+
+  # <mods:location><mods:url access="preview">
+  preview :class => DPLA::MAP::WebResource do
+    uri record.field('mods:location', 'mods:url')
+         .match_attribute(:access, 'preview')
+    dcformat record.field('mods:physicalDescription', 'mods:internetMediaType')
+  end
+
+  # <mods:location><mods:physicalLocation>
+  dataProvider :class => DPLA::MAP::Agent do
+    providedLabel record.field('mods:location', 'mods:physicalLocation')
+  end
+
+  # <mods:location><mods:url usage="primary" access="object in context">
+  isShownAt :class => DPLA::MAP::WebResource do
+    uri record.field('mods:location', 'mods:url')
+         .match_attribute(:usage, 'primary')
+         .match_attribute(:access, 'object in context')
+  end
+
+  sourceResource :class => DPLA::MAP::SourceResource do
+    alternative record.field('mods:titleInfo')
+                      .match_attribute(:type, 'alternative')
+                      .field('mods:title')
+
+    ##
+    # FIXME: The comment below was taken from the ESDN mapping.  Need to review
+    # the situation here.
+    #
+    # TODO: implement collection/set harvester and enrichment to complete
+    # the metadata required for collections. This just grabs the set's
+    # identifier from the OAI-PMH setSpec in the record header.
+    collection :class => DPLA::MAP::Collection,
+               # setSpec is set_spec due to a bug in Krikri;
```

Comment:
I believe this was fixed by https://github.com/dpla/KriKri/commit/fe57738647a0b060bd47d0bfdf0f67e50c2b9238#diff-8d0c691718e8aa47ff13511391860fb4

We're now using the harvested OAI records exactly, since we added support for that to Ruby OAI.